File: chapter5/ch5_pe5.c
#include <stdio.h>

int x;

void test() {
  x = 21;
  printf("%d", x);
  int x;
  printf("%d", x);
  x = 42;
  printf("%d", x);
}

int main() {
  test();
  printf("%d", x);
  return 0;
}

File: chapter5/Ch5Pe5.java
import java.lang.System;
public class Ch5Pe5 {
   public static int x = 0;
   private static void test() {
      System.out.println(x);
      x = 21;
      System.out.println(x);
      int x;
      x = 42;
      System.out.println(x);
   }
   public static void main(String [] args) {
      test();
      System.out.println(x);
   }
}
